### PR TITLE
User registration

### LINKED ISSUE #10: User Story 10, User Registration
As a visitor
When I click on the 'register' link in the nav bar
Then I am on the user registration page ('/register')
And I see a form where I input the following data:
- my name
- my street address
- my city
- my state
- my zip code
- my email address
- my preferred password
- a confirmation field for my password

When I fill in this form completely,
And with a unique email address not already in the system
My details are saved in the database
Then I am logged in as a registered user
I am taken to my profile page ("/profile")
I see a flash message indicating that I am now registered and logged in

### LINKED ISSUE #11: User Story 11, User Registration Missing Details
As a visitor
When I visit the user registration page
And I do not fill in this form completely,
I am returned to the registration page
And I see a flash message indicating that I am missing required fields

### LINKED ISSUE #12: User Story 12, Registration Email must be unique
As a visitor
When I visit the user registration page
If I fill out the registration form
But include an email address already in the system
Then I am returned to the registration page
My details are not saved and I am not logged in
The form is filled in with all previous data except the email field and password fields
I see a flash message telling me the email address is already in use

### LINKED ISSUE #78: Top and Bottom 5 not showing on Heroku, but pass in tests
Screen Shot 2020-11-03 at 2 12 03 PM

### PR DESCRIPTION
**Pull request by JoseLopez235**
PR Contains:
-Email error that redirects the user to change email
-All Test Passing
- closes #10 
- closes #11 
- closes #12

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class UsersController < ApplicationController
+  def new
+
+  end
+
+  def create
+    user = User.new(user_params)
+    @params = user_params
```
Comment (c-bartell) [inline app/controllers/users_controller.rb:8]: We might be able to move this to line 13 so that the variable is only set when it's needed.

**Review by c-bartell — APPROVED** [review]
It looks good to me!

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,40 @@
+<%= content_tag :h1, "User Registration" %>
+
+<%= form_tag "/register", method: :post do %>
```
Comment (KieraAllen) [inline app/views/users/new.html.erb:3]: I wonder if there's a way to shorten this or condense all the tags...

**Review by KieraAllen — APPROVED** [review]

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    if user.save
+      flash[:success] = "You are now registered and logged in."
+      redirect_to '/profile'
+    else
+      flash[:errors] = user.errors.full_messages.uniq.to_sentence
```
Comment (stillsheryl) [inline app/controllers/users_controller.rb:13]: Nice job on this message!

File: app/views/users/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,40 @@
+<%= content_tag :h1, "User Registration" %>
```
Comment (stillsheryl) [inline app/views/users/new.html.erb:1]: Nice use of ruby here

**Review by stillsheryl — APPROVED** [review]
Looks great!

**Pull request merged**